solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/ModelQuery.java

```diff
@@ -151,48 +160,125 @@ private boolean equalsTo(ModelQuery other) {
   public SolrQueryRequest getRequest() {
     return request;
   }
-
-  @Override  
-  public ModelWeight createWeight(IndexSearcher searcher, boolean needsScores, float boost) throws IOException {
-
+  
+  @Override
+  public ModelWeight createWeight(IndexSearcher searcher, boolean needsScores, float boost)
+      throws IOException {   
     final Collection<Feature> modelFeatures = meta.getFeatures();
     final Collection<Feature> allFeatures = meta.getAllFeatures();
-
     int modelFeatSize = modelFeatures.size();
     int allFeatSize = this.extractAllFeatures ? allFeatures.size() : modelFeatSize;
-
+    
```

Comment:
Can probably delete allFeatSize now, and assign extractedFeatureWeights to be of size features.size()